Examples/Algorithms/Alignment/src/AlignmentAlgorithm.cpp

```diff
@@ -0,0 +1,123 @@
+      ctx.eventStore.get<ProtoTrackContainer>(m_cfg.inputProtoTracks);
+  const auto& initialParameters = ctx.eventStore.get<TrackParametersContainer>(
+      m_cfg.inputInitialTrackParameters);
+
+  // Consistency cross checks
+  if (protoTracks.size() != initialParameters.size()) {
+    ACTS_FATAL("Inconsistent number of proto tracks and parameters");
+    return ProcessCode::ABORT;
+  }
+
+  size_t numTracksUsed = protoTracks.size();
+  if (m_cfg.maxNumTracks > 0 and
+      m_cfg.maxNumTracks < static_cast<int>(protoTracks.size())) {
+    numTracksUsed = m_cfg.maxNumTracks;
+  }
+
+  // Prepare the input track collection
+  std::vector<std::vector<IndexSourceLink>> sourceLinkTrackContainer;
+  sourceLinkTrackContainer.reserve(numTracksUsed);
+  std::vector<IndexSourceLink> trackSourceLinks;
+  for (std::size_t itrack = 0; itrack < numTracksUsed; ++itrack) {
+    // The list of hits and the initial start parameters
+    const auto& protoTrack = protoTracks[itrack];
+
+    // Clear & reserve the right size
+    trackSourceLinks.clear();
+    trackSourceLinks.reserve(protoTrack.size());
+
+    // Fill the source links via their indices from the container
+    for (auto hitIndex : protoTrack) {
+      auto sourceLink = sourceLinks.nth(hitIndex);
+      if (sourceLink == sourceLinks.end()) {
+        ACTS_FATAL("Proto track " << itrack << " contains invalid hit index"
+                                  << hitIndex);
+        return ProcessCode::ABORT;
+      }
+      trackSourceLinks.push_back(*sourceLink);
+    }
+    sourceLinkTrackContainer.push_back(trackSourceLinks);
+  }
+
+  // Prepare the output data with MultiTrajectory
+  TrajectoriesContainer trajectories;
+  trajectories.reserve(numTracksUsed);
+
+  // Construct a perigee surface as the target surface for the fitter
+  auto pSurface = Acts::Surface::makeShared<Acts::PerigeeSurface>(
+      Acts::Vector3{0., 0., 0.});
+
+  // Set the KalmanFitter options
+  TrackFitterOptions kfOptions(
+      ctx.geoContext, ctx.magFieldContext, ctx.calibContext,
+      MeasurementCalibrator(measurements), Acts::VoidOutlierFinder(),
+      Acts::LoggerWrapper{logger()}, Acts::PropagatorPlainOptions(),
+      &(*pSurface));
+
+  // Set the alignment options
+  ActsAlignment::AlignmentOptions<TrackFitterOptions> alignOptions(
+      kfOptions, m_cfg.alignedTransformUpdater, Acts::LoggerWrapper{logger()},
+      m_cfg.alignedDetElements, m_cfg.chi2ONdfCutOff, m_cfg.deltaChi2ONdfCutOff,
+      m_cfg.maxNumIterations);
+
+  ACTS_DEBUG("Invoke track-based alignment with " << numTracksUsed
+                                                  << " input tracks");
+  auto result =
+      m_cfg.align(sourceLinkTrackContainer, initialParameters, alignOptions);
+  if (result.ok()) {
+    ACTS_VERBOSE(
+        "Alignment finished with deltaChi2 = " << result.value().deltaChi2);
+  } else {
+    ACTS_WARNING("Alignment failed with " << result.error());
+  }
+
+  // @TODO: output aligned parameters
```

Comment:
You should just write them to the `WhiteBoard` here - then we decoupage output from running.

Reply:
Done.

Reply:
Good, thanks.